sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDestinationCheckpointData.cs

```diff
@@ -16,51 +16,90 @@ internal class BlobDestinationCheckpointData : BlobCheckpointData
         /// <summary>
         /// The content headers for the destination blob.
         /// </summary>
-        public BlobHttpHeaders ContentHeaders;
-        private byte[] _contentTypeBytes;
-        private byte[] _contentEncodingBytes;
-        private byte[] _contentLanguageBytes;
-        private byte[] _contentDispositionBytes;
-        private byte[] _cacheControlBytes;
+        public DataTransferProperty<string> CacheControl;
+        public bool PreserveCacheControl;
+        public byte[] CacheControlBytes;
+
+        public DataTransferProperty<string> ContentDisposition;
+        public bool PreserveContentDisposition;
+        public byte[] ContentDispositionBytes;
+
+        public DataTransferProperty<string> ContentEncoding;
+        public bool PreserveContentEncoding;
+        public byte[] ContentEncodingBytes;
+
+        public DataTransferProperty<string> ContentLanguage;
+        public bool PreserveContentLanguage;
+        public byte[] ContentLanguageBytes;
+
+        public DataTransferProperty<string> ContentType;
+        public bool PreserveContentType;
+        public byte[] ContentTypeBytes;
 
         /// <summary>
         /// The access tier of the destination blob.
         /// </summary>
-        public AccessTier? AccessTier;
+        public DataTransferProperty<AccessTier?> AccessTier;
+        public bool PreserveAccessTier;
 
         /// <summary>
-        /// The metadate for the destination blob.
+        /// The metadata for the destination blob.
         /// </summary>
-        public Metadata Metadata;
-        private byte[] _metadataBytes;
+        public DataTransferProperty<Metadata> Metadata;
+        public bool PreserveMetadata;
+        public byte[] MetadataBytes;
 
         /// <summary>
         /// The Blob tags for the destination blob.
         /// </summary>
-        public Tags Tags;
-        private byte[] _tagsBytes;
+        public DataTransferProperty<Tags> Tags;
+        public bool PreserveTags;
+        public byte[] TagsBytes;
 
         public override int Length => CalculateLength();
 
         public BlobDestinationCheckpointData(
             BlobType blobType,
-            BlobHttpHeaders contentHeaders,
-            AccessTier? accessTier,
-            Metadata metadata,
-            Tags blobTags)
+            DataTransferProperty<string> contentType,
+            DataTransferProperty<string> contentEncoding,
+            DataTransferProperty<string> contentLanguage,
+            DataTransferProperty<string> contentDisposition,
+            DataTransferProperty<string> cacheControl,
+            DataTransferProperty<AccessTier?> accessTier,
+            DataTransferProperty<Metadata> metadata,
+            DataTransferProperty<Tags> tags)
             : base(DataMovementBlobConstants.DestinationCheckpointData.SchemaVersion, blobType)
```

Comment:
Since we released version 1 now, we probably need to bump the schema version for this to 2. Additionally, we'll have to put in checks to fail if we see version 1 now (since we're ok failing for old preview versions).

Reply:
Upped to 2 and also add tests for this.